hw_classes/classes_hw.py

```diff
@@ -0,0 +1,265 @@
+
+class StrangeFloat(float):
+    def __getattr__(self, name):
+        func = f'super(StrangeFloat, self).__{name[:3]}__(float({name[name.find("_")+1:]}))'
+        try:
+            s = eval(func)
+            return StrangeFloat(s)
+        except AttributeError as ae:
+            for fun in dir(float)[::-1]:
+                if name[:3] in fun:
+                    f = f'super(StrangeFloat, self).{fun}(float({name[name.find("_")+1:]}))'
+                    s = eval(f)
+                    return StrangeFloat(s)
+
+                
+# Task 4 ---------------------------------------------------------------------------------------------------------
+ 
+    
+np = __import__('numpy')
+
+matrix = list.__new__(list)
+idx = int.__int__(0)
+while idx.__lt__(100):
+    matrix.__iadd__([list.__call__(range(idx, idx.__add__(10)))])
+    idx = idx.__add__(10)
+    
+selected_columns_indices = list.__call__((filter(lambda x: x in range(1, 5, 2), range(matrix.__len__()))))
+selected_columns = map(lambda x: list.__call__(x.__getitem__(col) for col in selected_columns_indices), matrix)
+
+arr = np.array(list.__call__(selected_columns))
+
+mask = arr[:, 1].__mod__(3).__eq__(0)
+
+new_arr = arr.__getitem__(mask)
+
+product = new_arr.__matmul__(new_arr.T)
+
+if (product.__getitem__(0).__lt__(1000)).all().__and__((product.__getitem__(2).__gt__(1000)).any()):
+    print(product.mean().__str__())
+
+    
+# Task 5 ---------------------------------------------------------------------------------------------------------
+ 
+    
+from abc import ABC, abstractmethod
+from Bio.SeqUtils.ProtParam import ProteinAnalysis as PA
+
+
+class BiologicalSequence(ABC):
+    def __init__(self, seq):
+        # we will store input sequence as a string
+        if isinstance(seq, list):
+            self.seq = ''.join(seq).upper()
+        else:
+            self.seq = seq.upper()
+    
+    def __len__(self):
+        # returns the length of the sequence, call: len()
+        return len(self.seq)
+        
+    def __getitem__(self, slc):
+        # returns i-th element of the sequence, call: obj[i] OR
+        # slice from start to stop with step=step
+        if isinstance(slc, int):
+            return self.seq[slc]  
+        elif isinstance(slc, slice):
+            new_slice = slice(slc.start, slc.stop, slc.step) # can delete this line actually...
+            return self.seq[new_slice]
+    
+    def __str__(self):
+        # returns sequence as a string, call: str()
+        # since we're storing seq as a string - just printing it out
+        return self.seq
+    
+    def __repr__(self):
+        # prints out info about the sequence
+        print('Sequence: ')
+        return self.seq
+    
+    def check_alphabet(self):
+        # checking the sequence for the presence of inappropriate letters
+        if set(self.seq).issubset(self.letters_set):
+            print('Everything is fine!')
+        else:
+            raise TypeError("Sequence contains invalid letter")
```

Comment:
Это уже не абстрактный класс, так как у него нет абстрактных методов. Здесь не нужно было ничего реализовывать, требовалось только задать интерфейс
```suggestion
    @abstractmethod
    def __len__(self):
        pass
     
    @abstractmethod       
    def __getitem__(self, slc):
        pass
    
    @abstractmethod
    def __str__(self):
        pass
    
    @abstractmethod
    def __repr__(self):
        # prints out info about the sequence
        pass
    
    @abstractmethod
    def check_alphabet(self):
        pass
```